python/lsst/daf/butler/registry/collections/_base.py

```diff
@@ -469,6 +529,21 @@ def _insert_collection_chain_rows(
             }
             for position, child in enumerate(child_keys, starting_position)
         ]
+
+        # It's possible for the DB to raise an exception for the integers being
+        # out of range here.  The position column is only a 16-bit number.
+        # Even if there aren't an unreasonably large number of children in the
+        # collection, a series of many deletes and insertions could cause the
+        # space to become fragmented.
+        #
+        # If this ever actually happens, we should consider doing a migration
+        # to increase the position column to a 32-bit number.
+        # To fix it in the short term, you can re-write the collection chain to
+        # defragment it by doing something like:
+        # registry.setCollectionChain(
+        #     parent,
+        #     registry.getCollectionChain(parent)
+        # )
```

Comment:
Not sure whether me not doing anything to address this is "efficient" or merely "lazy".  The most realistic scenario where this could happen is someone trying to use a collection chain as a queue... after around 32,000 prepends this will happen if you only remove from the end of the chain.

I should probably just write a migration script to make that column 32-bit.